src/fsharp/DotNetFrameworkDependencies.fs

```diff
@@ -482,6 +482,162 @@ module internal FSharp.Compiler.DotNetFrameworkDependencies
+            yield "System.Reflection.TypeExtensions"
+            yield "System.Resources.Reader"
+            yield "System.Resources.ResourceManager"
+            yield "System.Resources.Writer"
+            yield "System.Runtime.CompilerServices.Unsafe"
+            yield "System.Runtime.CompilerServices.VisualC"
+            yield "System.Runtime"
+            yield "System.Runtime.Extensions"
+            yield "System.Runtime.Handles"
+            yield "System.Runtime.InteropServices"
+            yield "System.Runtime.InteropServices.RuntimeInformation"
+            yield "System.Runtime.Intrinsics"
+            yield "System.Runtime.Loader"
+            yield "System.Runtime.Numerics"
+            yield "System.Runtime.Serialization"
+            yield "System.Runtime.Serialization.Formatters"
+            yield "System.Runtime.Serialization.Json"
+            yield "System.Runtime.Serialization.Primitives"
+            yield "System.Runtime.Serialization.Xml"
+            yield "System.Security.Claims"
+            yield "System.Security.Cryptography.Algorithms"
+            yield "System.Security.Cryptography.Csp"
+            yield "System.Security.Cryptography.Encoding"
+            yield "System.Security.Cryptography.Primitives"
+            yield "System.Security.Cryptography.X509Certificates"
+            yield "System.Security"
+            yield "System.Security.Principal"
+            yield "System.Security.SecureString"
+            yield "System.ServiceModel.Web"
+            yield "System.ServiceProcess"
+            yield "System.Text.Encoding.CodePages"
+            yield "System.Text.Encoding"
+            yield "System.Text.Encoding.Extensions"
+            yield "System.Text.Encodings.Web"
+            yield "System.Text.Json"
+            yield "System.Text.RegularExpressions"
+            yield "System.Threading.Channels"
+            yield "System.Threading"
+            yield "System.Threading.Overlapped"
+            yield "System.Threading.Tasks.Dataflow"
+            yield "System.Threading.Tasks"
+            yield "System.Threading.Tasks.Extensions"
+            yield "System.Threading.Tasks.Parallel"
+            yield "System.Threading.Thread"
+            yield "System.Threading.ThreadPool"
+            yield "System.Threading.Timer"
+            yield "System.Transactions"
+            yield "System.Transactions.Local"
+            yield "System.ValueTuple"
+            yield "System.Web"
+            yield "System.Web.HttpUtility"
+            yield "System.Windows"
+            yield "System.Xml"
+            yield "System.Xml.Linq"
+            yield "System.Xml.ReaderWriter"
+            yield "System.Xml.Serialization"
+            yield "System.Xml.XDocument"
+            yield "System.Xml.XmlDocument"
+            yield "System.Xml.XmlSerializer"
+            yield "System.Xml.XPath"
+            yield "System.Xml.XPath.XDocument"
+            yield "WindowsBase"
```

Comment:
Are WindowsBase/System.Windows/Microsoft.Win32.Primitives valid for non-windows FSI sessions?

Similarly, should all three 'primary assemblies' of mscorlib, System.Runtime, and netstandard.dll be referenced?

Reply:
see response below, thanks, it's a good question